src/util.rs

```diff
@@ -312,6 +313,90 @@ pub async fn unzip_file(
     Ok(ret)
 }
 
+pub async fn zip_files(
+    files: Vec<impl AsRef<Path>>,
+    dest: impl AsRef<Path>,
+    overwrite_old: bool,
+) -> Result<PathBuf, Error> {
+    let files = files.into_iter().map(|f| f.as_ref().to_path_buf()).collect::<Vec<PathBuf>>();
```

Comment:
this call will allocate, if you need ownership of the paths it's best to put it so in the function signature